### DISCUSSION TITLE
New typescript typings for 3.x

### DISCUSSION THREAD

**Discussion by seanforyou23** [TanStack/query#2368] [Q&A]
Hello!

I'm just getting going with this library and have been tasked with upgrading an app using react-query 2.x to 3.x. A lot of the upgrade has gone smooth so far, however we make use of several typescript types which are no longer available, such as

- QueryResult
- MutationConfig
- MutationResultPair
- MutationResult
- UseQueryObjectConfig

I've been through the upgrade guide and searched around in the type declaration files and other github issues, but not finding a suitable replacement for these types for the new version. Do any of these have a corresponding type in the new version? Or, in case I've missed a guide somewhere that explains how to best move forward with managing the type updates, could someone please point me there?

Many thanks in advance!

**Comment by TkDodo** [comment]
I try to let react-query infer most types and rarely use them at all. I’ve written about it here: 

https://tkdodo.eu/blog/react-query-and-type-script

Most types are defined here: https://github.com/tannerlinsley/react-query/blob/e42cbc32dfcd9add24cadd06135e42af1dbbc8ad/src/react/types.ts

so there is `UseQueryResult`, `UseQueryOptions` etc

**Reply by seanforyou23** [reply]
This looks really helpful, I'll have a look. Thank you!
